src/shared/defaultAwsContext.ts

```diff
@@ -105,9 +122,11 @@ export class DefaultAwsContext implements AwsContext {
     }
 
     private emitEvent() {
+        // TODO(jmkeyes): skip this if the state did not actually change.
```

Comment:
If you're feeling adventurous:
```ts
private createReactive<T extends Record<string, any>>(obj: T): T {
    return new Proxy(obj, {
        set: (target, prop, val, recv) => {
            const last = Reflect.get(target, prop, recv)
            const success = Reflect.set(target, prop, val, recv)

            if (success && last !== val) {
                this.emitEvent()
            }

            return success
        }
    })
}
```

Reply:
cool! Definitely we should revisit that in a followup PR.